### ISSUE TITLE
[Suggestion] Character sheet should be built off some kind of markup

### ISSUE THREAD

**Issue by dougrich** [kgrubb/microluxe20#10]
Take your pick of LaTeX or HTML, but right now it is checked in as a binary that can't be edited.

I've got an idea for HTML and could plug it into the build process if you'd be interested in seeing a pull request.

**Comment by kgrubb** [comment]
I'd much prefer html over LaTeX. I'm definitely interested. Having a dynamic character sheet would make major handbook changes much easier to implement.

**Comment by dougrich** [comment]
Alright, this is a sample rendered PDF so far: [character-sheet.pdf](https://github.com/kgrubb/microluxe20/files/79470/character-sheet.pdf)

To do:

- Text everywhere
- Fix alignment in top right
- Put symbols in around AC box and coinage boxes (probably inline svg)

The layout is determined by a set of scss variable declarations so it's really easy to change things like margin widths.

Once the text is gone I might also experiment with removing some of the boxes because it feels very boxy right now.

**Comment by kgrubb** [comment]
Any update regarding the character sheet setup? I'd love to take a look at what you've done so far. The sample looks very promising.

**Comment by Web-eWorks** [comment]
I don't have the means to do it in HTML, but if SVG is an acceptable compromise I'd be willing to take this on when I get some free time. Assuming, of course, that @dougrich is unable to continue with the original idea.

**Comment by kgrubb** [comment]
Go for it! Any automated solution is fine for now. If we want to go another route in the future we can refactor later.

**Comment by Web-eWorks** [comment]
Well, it wouldn't be automated per-se (unless I'm missing the point), but it would fix the uneditable binary bit. Will move forward on this soon.

**Comment by Web-eWorks** [comment]
So I took a crack at recreating the current sheet in SVG, but then I remembered that I could dump the PDF directly to SVG. After some mucking about with fonts, I got a reasonable result back out. If you want, I can PR it into the repo, but there's very little (if any) advantage over the PDF, which is actually quite editable in Draw.

I'll probably switch tacks and see about the automated HTML solution (with a slightly simpler layout), because I want a character sheet auto-generator that takes in JSON data and spits out rendered character sheets.

**Comment by kgrubb** [comment]
I've been working on a website update to allow for character creation online. I'm looking into using angular to allow people to submit character info and get a printable pdf. Sounds like this may be fairly in-line with the json idea.

**Comment by Web-eWorks** [comment]
Yeah, something like web UI -> JSON object + REST/Whatever -> Node task running serverside -> printable PDF returned to client. That about right, or am I totally off base?
